### ISSUE TITLE
Support "JavaScript Debug Terminal" from VS Code in Deno inspector

### ISSUE THREAD

**Issue by lucacasonato** [denoland/deno#19328]
https://code.visualstudio.com/docs/nodejs/nodejs-debugging#_javascript-debug-terminal

It seems to inject a `VSCODE_INSPECTOR_OPTIONS` and some `NODE_OPTIONS` env vars that are used to start and attach the inspector.

**Comment by cdaringe** [comment]
@lucacasonato , the debug terminal is the best!

- We don't have a `DENO_OPTIONS` in deno-land to set something like `--inspect-brk`. Has there been any discussion of supporting it? 
- Perhaps on "Debug Terminal Launch" for deno, the instrumentation could instead squash `deno` on the $PATH to point to an entry that bootstraps `deno` cli invocations, adding the necessary `--inspect-brk:PORTs` intelligently? That can get a little messy, as we'd need to be shell-aware. Part of the beauty of the debug terminal is that is also instruments _child process_ attaching, which is very powerful. Getting that to work would also require re-working the entry point script. e.g. if i run 
  - `deno run ./foo.ts`, that would actually run a custom entrypoint:
  - `$DENO_DIR/.debug/session-<uuid>/deno run ./foo.ts`, and that custom entrypoint perhaps inject some flags, a la
  - `/actual/path/to/deno run --inspect-brk:DERIVED_PORT  ./foo.ts`. `DERIVED_PORT` is as such s.t. child processes compute another port, and the IDE could tap into each.

Maybe the vscode folks would have recommendations or time for helping spec it out for us?

**Comment by cdaringe** [comment]
TLDR, https://github.com/microsoft/vscode-js-debug/blob/f1fe28ba28aa338598772dd6211a010dc3deea87/src/typings/vscode-js-debug.d.ts#L19 would let us intercept something like the `env`, modify it with our `deno` entry point, and _bam_, user's who type `deno run` suddenly could be proxied thru us, for which we could `--inspect` away!

**Comment by felipecrs** [comment]
@stefnotch made some cool discoveries about this [here](https://github.com/denoland/vscode_deno/issues/885#issuecomment-2867651957), using Bun as base (yes, Bun supports this feature).

Looks like what needs to be implemented in Deno CLI is the support for `DENO_INSPECT` and `DENO_INSPECT_NOTIFY`, named after their Bun counterparts.

It would be awesome to see this feature for Deno, it's a huge deal for whoever develops CLI apps.

**Comment by bartlomieju** [comment]
This now works. VS Code's JavaScript Debug Terminal auto-attaches by injecting `NODE_OPTIONS` (with `--require <bootloader>` and `--inspect-publish-uid`) plus `VSCODE_INSPECTOR_OPTIONS` into the terminal environment. Deno now honors all the flags it needs from `NODE_OPTIONS`:

- `--require` / `-r` and `--inspect-publish-uid` — #31949
- `--inspect` / `--inspect-brk` / `--inspect-wait` — #34717

Verified on `main`: launching a script with the dev build inside a JavaScript Debug Terminal loads the injected bootloader and the debugger attaches and stops at breakpoints. Closing as resolved.
